### PR TITLE
add Feature and FeatureCollection validation from string

### PR DESCRIPTION
**Pull request by vincentsarago**
ref #69 

@Vikka, @geospatial-jeff could you double check that I'm not doing any wrong stuff here 🙏

### CODE REVIEW

File: geojson_pydantic/features.py [hunk 1]
```diff
@@ -62,3 +73,16 @@ def __getitem__(self, index):
...
+    @classmethod
+    def validate(cls, value):
+        """Validate input."""
+        try:
+            value = json.loads(value)
+        except TypeError:
+            try:
+                return cls(**value.dict())
+            except (AttributeError, ValidationError):
+                pass
```
Comment (Vikka) [inline geojson_pydantic/features.py:86]: I don't think that a FeatureCollection object will ever be validated. 
If it does, I don't know which path can lead to this behaviour, and a test for this is missing.

```suggestion
            pass
```

**Review by Vikka — CHANGES_REQUESTED** [review]
I think that the validate function is hard to understand for a newcommer, and I'm currently working on an update of the code, but python >= 3.8 I'm affraid. I will do a draft when this PR will be merged, to show you what I'm suggesting to clarify the validate function.

File: geojson_pydantic/features.py (hunk 1 above)
Comment (vincentsarago) [reply geojson_pydantic/features.py:86]: 🤷‍♂️ I don't really understand, I `copy/pasted` this to follow the same logic as what you did for GeometryCollection

Comment (Vikka) [reply geojson_pydantic/features.py:86]: Ok I see. Then the issue is on `Feature`. I'll comment it.

File: geojson_pydantic/features.py [hunk 2]
```diff
@@ -38,6 +39,16 @@ def __geo_interface__(self):
         """GeoJSON-like protocol for geo-spatial (GIS) vector data."""
         return self.dict()
 
+    @classmethod
+    def validate(cls, value):
+        """Validate input."""
+        try:
+            value = json.loads(value)
+        except TypeError:
+            pass
```
Comment (Vikka) [inline geojson_pydantic/features.py:48]: Why not validation of `BaseModel` objects?

```suggestion
            try:
                return cls(**value.dict())
            except (AttributeError, ValidationError):
                pass
```

**Review by Vikka — CHANGES_REQUESTED** [review]
If we agree that `FeatureCollection` should validate `BaseModel` objects (like `Geometry`), then I think `Feature` should too.

**Comment by vincentsarago** [comment]
@Vikka any else missing here?

**Review by Vikka — APPROVED** [review]

**Pull request merged**